Ccg SUSE Linux Enterprise Server with Expanded Support 9 #19568 keichwa

keichwa opened the pull request:
# Description

SUSE Linux Enterprise Server with Expanded Support 9:

https://github.com/SUSE/spacewalk/issues/19568
https://github.com/uyuni-project/uyuni-docs/issues/1897
https://github.com/SUSE/spacewalk/issues/19159 (?)

# Target branches

Which documentation version does this PR apply to?

- [x] Master (Default)
- [ ] Manager-4.3
- [ ] Manager-4.2



# Links

Fixes #<insert issue or PR link, if any>

lucidd commented on modules/client-configuration/pages/clients-sleses.adoc:
```diff
@@ -95,6 +98,7 @@ This procedure creates these tools channels:
 | OS Version | Base Channel                     | Tools Channel
 | {es} 7     | RHEL Expanded Support 7          | RES7-SUSE-Manager-Tools x86_64
 | {es} 8     | RHEL or SLES ES or CentOS 8 Base | RES8-Manager-Tools-Pool for x86_64 and RES8-Manager-Tools-Updates for x86_64
+| {es} 9     | RHEL or SLES ES                  | RES9-Manager-Tools-Pool for x86_64 and RES9-Manager-Tools-Updates for x86_64
```
according tho https://github.com/SUSE/spacewalk/issues/19085#issuecomment-1318541475 the base channel will be named "RHEL, Liberty and Clones Base" right now its still called "EL 9 Base"

lucidd replied:
The description should probably also be adjusted to reflect the changed naming

keichwa replied:
ok, I'll now make use of the future name with Liberty.

lucidd commented on modules/client-configuration/pages/clients-sleses.adoc:
```diff
@@ -134,6 +138,11 @@ The channels you need for this procedure are:
 | res8-manager-tools-pool-x86_64
 
+| {es} 9
+| rhel8-pool-x86_64
```
pool label is `el9-pool-x86_64` and tools label is `el9-manager-tools-pool-x86_64`

keichwa replied:
ok, will change it.

keichwa commented:
Kevin (@lucidd), I hope I got the names and labels right.  Could we still keep the "Expanded Support" name in parallel with Liberty 9 for "{es} 9"? Please, review again!

admd commented:
We should start replacing RES /SLES ES with Liberty now where possible.

We should add a note in one single place though that all RES /SLES ES would be mentioned as Liberty from now on.

keichwa commented:
> We should start replacing RES /SLES ES with Liberty now where possible. No need to keep all.
> 
> We should add a note in one single place though that all RES /SLES ES would be mentioned as Liberty from now on.

Ok, makes sense.  But the channel names and labels of old versions will stay as they are, will they? I think I better ask twice...

admd commented:
> > We should start replacing RES /SLES ES with Liberty now where possible. No need to keep all.
> > We should add a note in one single place though that all RES /SLES ES would be mentioned as Liberty from now on.
> 
> Ok, makes sense. But the channel names and labels of old versions will stay as they are, will they? I think I better ask twice...

yes channel labels will stay the same.

lucidd commented:
just an update on this i changed the name to `RHEL and Liberty 9 Base` since clones is not really accurate in this case everything else stays the same.

keichwa commented:
Kevin Walter ***@***.***> writes:

> just an update on this i changed the name to `RHEL and Liberty 9 Base`
> since clones is not really accurate in this case everything else stays
> the same.

Thanks, sounds good!  I fixed the docs.

-- 
Karl Eichwalder                       SUSE Software Solutions Germany GmbH
                                      Frankenstraße 146, D-90461 Nürnberg
Geschäftsführer: Ivo Totev, Andrew Myers, Andrew McDonald, Boudien Moerman
(HRB 36809, AG Nürnberg)

keichwa commented:
[suse_manager_client-configuration_guide.pdf](https://github.com/uyuni-project/uyuni-docs/files/10110320/suse_manager_client-configuration_guide.pdf)
[uyuni_client-configuration_guide.pdf](https://github.com/uyuni-project/uyuni-docs/files/10110323/uyuni_client-configuration_guide.pdf)

srbarrios commented on modules/client-configuration/pages/clients-sleses.adoc:
```diff
@@ -93,16 +101,17 @@ This procedure creates these tools channels:
-| {es} 8     | RHEL or SLES ES or CentOS 8 Base | RES8-Manager-Tools-Pool for x86_64 and RES8-Manager-Tools-Updates for x86_64
+| {sll} 7     | RHEL Expanded Support 7          | RES7-SUSE-Manager-Tools x86_64
+| {sll} 8     | RHEL or SLES ES or CentOS 8 Base | RES8-Manager-Tools-Pool for x86_64 and RES8-Manager-Tools-Updates for x86_64
+| {sll} 9     | RHEL and Liberty 9 Base          | RES9-Manager-Tools-Pool for x86_64 and RES9-Manager-Tools-Updates for x86_64
```
Are we sure we use RES9 in the tools channel? I thought it was renamed to EL9

lucidd replied:
the update channel is on the liberty linux extension product and named `SLL-9-Updates for x86_64`

keichwa replied:
https://suma-bv-43-srv.mgr.prv.suse.net/rhn/manager/admin/setup/products has an unfinished installation since a week or longer.  I'll now use
```
| {sll} 9     | RHEL and Liberty 9 Base          | EL9-Manager-Tools-Pool for x86_64 and SLL9-Manager-Tools-Updates for x86_64
```
If this is not correct please provide a c&p variant.

lucidd replied:
the channel names as they are right now on my fresh setup are `EL9-Manager-Tools-Pool for x86_64` and `EL9-Manager-Tools-Updates for x86_64`

srbarrios replied:
@lucidd @keichwa should we mention el8-uyuni-client-x86_64, el9-uyuni-client-x86_64 somewhere? Please give another look to this page. Still seems not good.
For more context, follow this thread https://suse.slack.com/archives/C02D12TNYLS/p1673872934090469?thread_ts=1673872675.415999&cid=C02D12TNYLS please

mcalmer replied:
The uyuni channels cannot be added with this page. You can only add them with spacewalk-common-channels tool which is described for uyuni. AFAIK we have this there.

srbarrios replied:
https://bugzilla.suse.com/show_bug.cgi?id=1205719#c4

srbarrios commented on modules/client-configuration/pages/clients-sleses.adoc:
```diff
@@ -182,7 +196,7 @@ If you do not provide Python 3.6, the installation of the [package]``spacecmd``
 . On the {productname} Server, at the command prompt, as root, copy the base media image to the [path]``/tmp/`` directory.
 . Create a directory to contain the media content.
-    Replace [command]``<os_name>`` with either ``sleses6``,  ``sleses7``, or ``sleses8``:
+    Replace [command]``<os_name>`` with either ``sleses7``, ``sleses8``, or ``sleses9``:
```
sleses? Are we sure about it?

keichwa replied:
ok.  done.

Bischoff commented on modules/client-configuration/pages/clients-sleses.adoc:
```diff
@@ -1,13 +1,18 @@
+{sll} clients are sometimes also called {sleses} (SLESES), Liberty, RES or {redhat} {es}.
+====
+
+The {sll} software channels provided by {suse} only provide updates to packages, they do not provide the packages themselves.
```
Is that still true ? Now SLL is a "real" product, that can be installed. Is that to say we have no "pool" packages ?

admd replied:
Yes, this is still true. Only the updates. SUSE don't provide any pool channel from scc when it comes to SLL.

keichwa replied:
ok, thanks for clarification!

Bischoff reviewed (commented):
see comments

keichwa commented:
@lucidd and @juliogonzalez , can you please re-check.  I'll upload PDF output later.

keichwa commented:
I ping'ed devs for final clarification.

keichwa commented:
Mayby, we also still depend on https://github.com/SUSE/spacewalk/issues/19159

srbarrios commented:
> Mayby, we also still depend on [SUSE/spacewalk#19159](https://github.com/SUSE/spacewalk/issues/19159)

Personally I would do it in two PRs. As SLL9 support could take some time yet.

keichwa commented on modules/client-configuration/pages/clients-sleses.adoc:
```diff
@@ -66,21 +71,24 @@ The products you need for this procedure are:
 //| SUSE Linux Enterprise Server with Expanded Support 6 x86_64
 
-| {es} 7
+| {sll} 7
```
In https://www.suse.com/lifecycle/, it is still listed.  OTOH, there you'll see the string "TBD".

keichwa commented on modules/client-configuration/pages/clients-sleses.adoc:
```diff
@@ -156,8 +170,8 @@ include::snippets/appstream_admon.adoc[]
 // 2022-04-26, ke: Here we can keep the "jeos" abbrev (rh product line)
-The base {es} channel does not contain any packages, because {suse} does not provide {rhel} or {centos} base media.
-You need to obtain base media from {redhat} or {centos}, which you can add as a child channel to the {es} parent channel.
+The base {sll} channel does not contain any packages, because {suse} does not provide {rhel} or {centos} base media.
```
I guess we'll better work on this separately.  Let's merge it first.

keichwa commented:
![EL9-Updates](https://user-images.githubusercontent.com/1759421/211290085-e557ecd0-1d89-4ef6-b77a-e645c89705d7.png)
